tt_metal/impl/device/multi_device.cpp

```diff
@@ -23,43 +22,46 @@ DeviceMesh::DeviceMesh(const DeviceGrid& device_grid, const DeviceIds &device_id
     TT_ASSERT(num_requested_devices <= num_available_devices, "Requested more devices than available");
     TT_ASSERT(num_requested_devices <= device_ids.size(), "User provided insufficient number of device_ids for DeviceMesh");
 
-
-    //TODO: for DevicePool feature delete CreateDevices and merge with this function
-    //TODO: should there be an explicit CloseDevices call somewhere?
     bool is_galaxy = tt::Cluster::instance().is_galaxy_cluster();
-    std::vector<chip_id_t> mmio_device_ids = {};
-    if (is_galaxy) {
-        mmio_device_ids.push_back(0);
-        if (num_requested_devices > 8) {
-            mmio_device_ids.push_back(1);
-        }
-        if (num_requested_devices > 16) {
-            mmio_device_ids.push_back(2);
-        }
-        if (num_requested_devices > 24) {
-            mmio_device_ids.push_back(3);
-        }
-    } else {
-        mmio_device_ids = device_ids;
-    }
-    managed_devices = tt::tt_metal::detail::CreateDevices(mmio_device_ids, 1, l1_small_size, trace_region_size);
     if (is_galaxy) {
+        // Temp solution until we add algorithmic way to determine chip connectivity
+        // Map col to tunnel depth and row to tunnel count
+        int cluster_tunnel_depth = tt::Cluster::instance().get_mmio_device_max_tunnel_depth(0);
```

Comment:
Is the `0` supposed to be first reference mmio device?

Reply:
yes.

Reply:
Does this correctly support the `DeviceGrid(3, 2)` case where num_devices < 8 but I assume we need multiple MMIO devices?

Reply:
I think you would need at least 2 MMIO devices to support 3x2.